### PR TITLE
Use 3D coords in monster code

### PR DESCRIPTION
**Pull request by Coolthulhu**
Changed almost the entire monster class code to use tripoints instead of x/y ints.
- Plans
- Wander position
- Attack/move/bash commands
- Knock back from

Since I rewrote most of the monmove.cpp and anyone touching it would almost certainly get conflicts, I decided to also astyle it.

### CODE REVIEW

File: src/melee.cpp [hunk 1]
```diff
@@ -1194,11 +1194,12 @@ void player::perform_technique(ma_technique technique, Creature &t, int &bash_da
     }
 
     if (technique.knockback_dist > 0) {
-        int kb_offset = rng(
-                            -technique.knockback_spread,
-                            technique.knockback_spread
-                        );
-        t.knock_back_from(posx() + kb_offset, posy() + kb_offset);
+        const int kb_offset_x = rng( -technique.knockback_spread,
```
Comment (kevingranade) [inline src/melee.cpp]: Super nit-pick, this should ideally be a random distance and a random rotation based on rl_dist(), but I doubt the distances are high enough to make a difference in practice.

File: src/savegame_legacy.cpp [hunk 2]
```diff
@@ -2054,16 +2054,19 @@ std::istream& operator>>(std::istream& is, SkillLevel& obj) {
...
     int idtmp, plansize, speed, faction_dummy;
-    dump >> idtmp >> position.x >> position.y >> wandx >> wandy >> wandf >> moves >> speed >>
-         hp >> sp_timeout[0] >> plansize >> friendly >> faction_dummy >> mission_id >>
-         no_extra_death_drops >> dead >> anger >> morale;
+    dump >> idtmp >> position.x >> position.y >> wander_pos.x >> wander_pos.y >> wandf
+         >> moves >> speed >> hp >> sp_timeout[0] >> plansize >> friendly
+         >> faction_dummy >> mission_id >> no_extra_death_drops >> dead >> anger >> morale;
 
+    wander_pos.z = g->get_levz();
+    position.x = g->get_levz();
```
Comment (BevapDin) [inline src/savegame_legacy.cpp]: `x`?

Comment (Coolthulhu) [inline src/savegame_legacy.cpp]: Oops

File: src/savegame_legacy.cpp [hunk 3]
```diff
@@ -2054,16 +2054,19 @@ std::istream& operator>>(std::istream& is, SkillLevel& obj) {
...
     int idtmp, plansize, speed, faction_dummy;
-    dump >> idtmp >> position.x >> position.y >> wandx >> wandy >> wandf >> moves >> speed >>
-         hp >> sp_timeout[0] >> plansize >> friendly >> faction_dummy >> mission_id >>
-         no_extra_death_drops >> dead >> anger >> morale;
+    dump >> idtmp >> position.x >> position.y >> wander_pos.x >> wander_pos.y >> wandf
+         >> moves >> speed >> hp >> sp_timeout[0] >> plansize >> friendly
+         >> faction_dummy >> mission_id >> no_extra_death_drops >> dead >> anger >> morale;
 
+    wander_pos.z = g->get_levz();
+    position.z = g->get_levz();
```
Comment (BevapDin) [inline src/savegame_legacy.cpp]: Turns out, position isn't a tripoint, it's a plain, old, simple point. Did you mean `zpos`?

Comment (Coolthulhu) [inline src/savegame_legacy.cpp]: Yes, I did. That was a blind quick fix (I'd have to recompile the whole game to test it and I'm still cleaning up AIM).

**Pull request merged**